src/components/auth/CompanyForm.tsx

```diff
@@ -1,56 +1,132 @@
-import {View} from 'react-native';
+import {TouchableOpacity, View} from 'react-native';
 import React from 'react';
+import * as Yup from 'yup';
+import {Formik} from 'formik';
+
 import TextComponent from '../TextComponent';
 import InputComponent from '../InputComponent';
-import useQuery from '../../hook/useQuery';
-import useAuth from '../../hook/useAuth';
 import {globalStyles} from '../../theme/globalTheme';
 
-const CompanyForm = () => {
-  const {user, setUser} = useQuery();
-
-  const {
-    email,
-    phone,
-    password,
-    setPassword,
-    setConfirmPass,
-    confirmPass,
-    handleEmailOrPhoneChange,
-  } = useAuth();
+const SigupCompanyScheme = Yup.object().shape({
+  name: Yup.string()
+    .min(4, 'Too short name')
+    .max(30, 'Too long name')
```

Comment:
Maybe try to change th writing on ths and the password, may someting like the name is too short
And how is a name too long yo're basically letting out a lot of peolple

Reply:
What if my company has more than 30 characters

Reply:
@juperameza is right, for example even the legal name of BrightCoders has 37 characters: 

> "BRIGHTCODERS CONSULTING, S.A. DE C.V."